pandas/io/formats/latex.py

```diff
@@ -107,89 +137,59 @@ def pad_empties(x):
 
             # Get rid of old multiindex column and add new ones
             strcols = out + strcols[1:]
+        return strcols
 
-        if self.column_format is None:
-            dtypes = self.frame.dtypes._values
-            column_format = "".join(map(get_col_type, dtypes))
-            if self.fmt.index:
-                index_format = "l" * self.frame.index.nlevels
-                column_format = index_format + column_format
-        elif not isinstance(self.column_format, str):  # pragma: no cover
-            raise AssertionError(
-                f"column_format must be str or unicode, not {type(column_format)}"
-            )
+    def _preprocess_row(self, row):
+        """Preprocess elements of the row."""
+        if self.fmt.escape:
+            crow = self._escape_symbols(row)
         else:
-            column_format = self.column_format
-
-        self._write_tabular_begin(buf, column_format)
+            crow = [x if x else "{}" for x in row]
+        if self.fmt.bold_rows and self.fmt.index:
+            crow = self._convert_to_bold(crow, self._ilevels)
+        return crow
 
-        buf.write("\\toprule\n")
+    @staticmethod
```

Comment:
instead of using staticmethods can you make these module level functions. I find this much harder to follow and non-idiomatic.

Reply:
Moved static methods to module level. My intention originally was to keep the methods in the class, since these functions are not to be used elsewhere, in general, and to make it clear where the functions belong to. Anyway, I did as you suggested.